ADC_DMA/Core/Src/main.c

```diff
@@ -0,0 +1,184 @@
+/*
+ * 	Implement a routine that reads the internal voltage reference (ADC_IN17)
+ * 	and internal temperature sensor (ADC_IN18).
+ * 	The internal reference voltage must be in millivolts and the temperature in Celsius.
+ *
+ * 	1 - Polling		[X]
+ * 	2 - Interrupt	[X]
+ * 	3 - DMA			[/]
+ */
+
+#include "stm32l053xx.h"
+#include "macros.h"
+
+typedef struct {
+	uint16_t	v_ref;			//internal reference voltage measured (after calculation)
+	int16_t		temperature;	//temperature in Celsius degrees measured (after calculation)
+}sADC_measure;	//struct with all the information read by the ADC channel
+
+void 		wait(uint16_t time);	//delay function, system clock based
+
+int main(void)
+{
+	sADC_measure 	ADC_measure = {0};	//struct with all the information read by the ADC channel
+	uint16_t 		aux = 0;			//we only use aux to calculate v_ref
+
+	/*							  *
+	 * --- DMA INITIALIZATION --- *
+	 *							  */
+	RCC->AHBENR |= RCC_AHBENR_DMAEN;
+	DMA1_Channel1->CPAR	= 	(uint32_t) &(ADC1->DR);		//setting peripheral address
+	DMA1_Channel1->CCR 	|=	DMA_CCR_CIRC	|			//DMA in circular mode
+							DMA_CCR_MINC	|			//memory increment
+							DMA_CCR_PSIZE_0	|			//peripheral size set to 16 bits
+							DMA_CCR_MSIZE_0	;			//memory size set to 16 bits
+
+	/*										*
+	 * --- ADC INITIALIZATION PROCEDURE --- *
+	 *										*/
+	RCC->APB2ENR 	|= 	RCC_APB2ENR_ADC1EN;		//enables ADC clock
+	ADC1->CR 		|=	ADC_CR_ADVREGEN;		//enables the voltage regulator
+	ADC1->CFGR2 	|= 	ADC_CFGR2_CKMODE_0;		//sets ADC clock as PCLK/2
+	ADC->CCR 		|= 	ADC_CCR_LFMEN;			//low frequency mode enable
+	ADC1->SMPR 		|= 	ADC_SMPR_SMPR_2;		//sets sample time for 19.5 ADC clock cycles (14 needed)
+	ADC1->CFGR1		|=	ADC_CFGR1_WAIT;			//enables wait mode to prevent overrun
+
+	ADC1->CR |= ADC_CR_ADCAL;					//starting the calibration
+	while(ADC1->CR & ADC_CR_ADCAL);				//we have to wait until ADCAL = 0 (Can be handled by interrupt)
+
+	ADC1->CFGR1	|=	ADC_CFGR1_DMACFG |			//DMA in circular mode
+					ADC_CFGR1_DMAEN	;			//enables DMA	- must be done AFTER calibrate the AD
+
+	ADC1->ISR|= ADC_ISR_ADRDY;					//clear the ADRDY bit by programming it to 1
+	ADC1->CR |= ADC_CR_ADEN;					//then we enable the ADC
```

Comment:
Just realised that you enable AD here and then disable on line 64. Is there a reason for it?